xmtp_mls/src/subscriptions/process_welcome.rs

```diff
@@ -293,22 +298,30 @@ where
             })
         );
 
-        if let Ok(_)
-        | Err(GroupError::ProcessIntent(ProcessIntentError::WelcomeAlreadyProcessed(_))) = res
+        let id = welcome.id as i64;
+        if let Ok(maybe_group) = res {
+            Ok(maybe_group)
+        } else if let Err(GroupError::ProcessIntent(ProcessIntentError::WelcomeAlreadyProcessed(
+            _,
+        ))) = res
         {
-            self.load_from_store(welcome.id as i64)
+            Ok(self.load_from_store(id)?)
         } else {
             Err(res.expect_err("Checked for Ok value").into())
         }
     }
 
     /// Load a group from disk by its welcome_id
-    fn load_from_store(&self, id: i64) -> Result<(MlsGroup<Context>, i64)> {
-        let group = self
-            .context
-            .db()
-            .find_group_by_welcome_id(id)?
-            .ok_or(NotFound::GroupByWelcome(id))?;
+    fn load_from_store(&self, id: i64) -> Result<Option<MlsGroup<Context>>> {
+        let maybe_group = self.context.db().find_group_by_welcome_id(id)?;
+        let Some(group) = maybe_group else {
+            tracing::warn!(
+                welcome_id = id,
+                "Already processed welcome not loaded from store (likely oneshot message)"
+            );
+            return Ok(None);
+        };
```

Comment:
FYI in case this causes issues @insipx, we now log instead of returning an error for already processed welcomes for which there is no group in the DB

Reply:
looks good! it's a nice catch, didn't notice that dm groups may look like they're missing when querying by welcome id